cl/corpus_importer/management/commands/make_aws_manifest_files.py

```diff
@@ -187,6 +210,341 @@ def get_custom_query(
+        timestamp: A datetime object representing the month for which to
+            retrieve records.
+        options: A dictionary containing any additional options.
+
+    Returns:
+        A list of unique integer IDs for the records of the specified type
+        that were created or modified on or after the given `timestamp`.
+    """
+    related_field = "pk"
+    main_model = None
+    nested_models: list[type[AbstractDateTimeModel]] = []
+    match record_type:
+        case SEARCH_TYPES.OPINION:
+            main_model = Opinion
+            nested_models = []
+        case SEARCH_TYPES.ORAL_ARGUMENT:
+            main_model = Audio
+            nested_models = []
+        case SEARCH_TYPES.PEOPLE:
+            main_model = Person
+            nested_models = [
+                Source,
+                ABARating,
+                Education,
+                PoliticalAffiliation,
+                Position,
+            ]
+            related_field = "person_id"
+        case "fd":
+            main_model = FinancialDisclosure
+            nested_models = [
+                Agreement,
+                Debt,
+                Gift,
+                Investment,
+                NonInvestmentIncome,
+                Reimbursement,
+                SpouseIncome,
+                FDPosition,
+            ]
+            related_field = "financial_disclosure_id"
+        case _:
+            raise NotImplementedError(
+                f"Record type '{record_type}' is not supported."
+            )
+
+    record_ids = get_records_modified_since(main_model, "pk", timestamp)
+    # check nested/related models
+    for model in nested_models:
+        record_ids.extend(
+            get_records_modified_since(model, related_field, timestamp)
+        )
+
+    match record_type:
+        case SEARCH_TYPES.OPINION:
+            # Apply filters to include only opinions that are either:
+            # 1. Not extracted by OCR.
+            # 2. Associated with a cluster whose source contains HARVARD_CASELAW
+            ids = [x[0] for x in record_ids]
+            return list(
+                Opinion.objects.filter(pk__in=ids)
+                .filter(
+                    Q(extracted_by_ocr=False)
+                    | Q(cluster__source__icontains=SOURCES.HARVARD_CASELAW)
+                )
+                .values_list("pk")
+                .distinct()
+            )
+        case SEARCH_TYPES.PEOPLE:
+            # Filter record IDs for people, including only those who have
+            # judicial positions.
+            ids = [x[0] for x in record_ids]
+            return list(
+                Person.objects.filter(pk__in=ids)
+                .filter(positions__position_type__in=JUDICIAL_POSITIONS)
+                .values_list("pk")
+                .distinct()
+            )
+        case _:
+            return list(set(record_ids))
+
+
+def compute_monthly_export(
+    record_type: str, timestamp: datetime, options: dict[str, Any]
```

Comment:
Also, `timestamp` and `options` differ from the names used in the docstring.

Reply:
Thanks!